### ISSUE TITLE
Filebeat 5.2.0 Released, Unable to Install

### ISSUE THREAD

**Issue by kmcarpenter** [vkhatri/chef-filebeat#88]
Looks like Filebeat 5.2.0 has been released and this recipe is now failing. I have tried overriding the version in my config to be 5.2.0 but this still seems to be not working.

```
Recipe: filebeat::install_package
  * apt_repository[beats] action add
    * remote_file[/root/.chef/local-mode-cache/cache/https___artifacts_elastic_co_GPG-KEY-elasticsearch] action create (up to date)
    * execute[apt-key add /root/.chef/local-mode-cache/cache/https___artifacts_elastic_co_GPG-KEY-elasticsearch] action run (skipped due to not_if)
    * execute[apt-cache gencaches] action nothing (skipped due to action :nothing)
    * apt_update[beats] action nothing (skipped due to action :nothing)
    * file[/etc/apt/sources.list.d/beats.list] action create
      - update content in file /etc/apt/sources.list.d/beats.list from 7fb750 to fcfb6b
      --- /etc/apt/sources.list.d/beats.list	2017-02-01 16:24:57.108262000 +0000
      +++ /etc/apt/sources.list.d/.chef-beats20170201-2907-1cz9slh.list	2017-02-01 16:27:59.069197001 +0000
      @@ -1,2 +1,2 @@
      -deb https://artifacts.elastic.co/packages/5.x/apt stable main
      +deb      "https://artifacts.elastic.co/packages/5.x/apt"  stable main
    * execute[apt-cache gencaches] action run
      - execute apt-cache gencaches
    * apt_update[beats] action update
      - force update new lists of packages
      * directory[/var/lib/apt/periodic] action create (up to date)
      * directory[/etc/apt/apt.conf.d] action create (up to date)
      * file[/etc/apt/apt.conf.d/15update-stamp] action create_if_missing (up to date)
      * execute[apt-get -q update] action run
        
        ================================================================================
        Error executing action `run` on resource 'execute[apt-get -q update]'
        ================================================================================
        
        Mixlib::ShellOut::ShellCommandFailed
        ------------------------------------
        Expected process to exit with [0], but received '100'
        ---- Begin output of apt-get -q update ----
        STDOUT: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
        Hit:2 http://ppa.launchpad.net/ondrej/php/ubuntu xenial InRelease
        Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
        Get:4 http://archive.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
        Fetched 204 kB in 0s (246 kB/s)
        Reading package lists...
        STDERR: E: The method driver /usr/lib/apt/methods/https could not be found.
        E: Failed to fetch https://artifacts.elastic.co/packages/5.x/apt/dists/stable/InRelease  
        E: Some index files failed to download. They have been ignored, or old ones used instead.
        ---- End output of apt-get -q update ----
        Ran apt-get -q update returned 100
        
        Resource Declaration:
        ---------------------
        # In /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_update.rb
        
         80:         declare_resource(:execute, "apt-get -q update")
         81:       end
        
        Compiled Resource:
        ------------------
        # Declared in /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_update.rb:80:in `do_update'
        
        execute("apt-get -q update") do
          action [:run]
          retries 0
          retry_delay 2
          default_guard_interpreter :execute
          command "apt-get -q update"
          backup 5
          returns 0
          declared_type :execute
          cookbook_name "filebeat"
        end
        
        Platform:
        ---------
        x86_64-linux
        
[2017-02-01T16:28:01+00:00] ERROR: apt_update[beats] (/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_repository.rb line 59) had an error: Mixlib::ShellOut::ShellCommandFailed: execute[apt-get -q update] (/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_update.rb line 80) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
---- Begin output of apt-get -q update ----
STDOUT: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
Hit:2 http://ppa.launchpad.net/ondrej/php/ubuntu xenial InRelease
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
Fetched 204 kB in 0s (246 kB/s)
Reading package lists...
STDERR: E: The method driver /usr/lib/apt/methods/https could not be found.
E: Failed to fetch https://artifacts.elastic.co/packages/5.x/apt/dists/stable/InRelease  
E: Some index files failed to download. They have been ignored, or old ones used instead.
---- End output of apt-get -q update ----
Ran apt-get -q update returned 100; ignore_failure is set, continuing
      
      ================================================================================
      Error executing action `update` on resource 'apt_update[beats]'
      ================================================================================
      
      Mixlib::ShellOut::ShellCommandFailed
      ------------------------------------
      execute[apt-get -q update] (/opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_update.rb line 80) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
      ---- Begin output of apt-get -q update ----
      STDOUT: Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
      Hit:2 http://ppa.launchpad.net/ondrej/php/ubuntu xenial InRelease
      Get:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
      Get:4 http://archive.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
      Fetched 204 kB in 0s (246 kB/s)
      Reading package lists...
      STDERR: E: The method driver /usr/lib/apt/methods/https could not be found.
      E: Failed to fetch https://artifacts.elastic.co/packages/5.x/apt/dists/stable/InRelease  
      E: Some index files failed to download. They have been ignored, or old ones used instead.
      ---- End output of apt-get -q update ----
      Ran apt-get -q update returned 100
      
      Resource Declaration:
      ---------------------
      # In /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_repository.rb
      
       59:         declare_resource(:apt_update, new_resource.name) do
       60:           ignore_failure true
       61:           action :nothing
       62:         end
       63: 
      
      Compiled Resource:
      ------------------
      # Declared in /opt/chefdk/embedded/lib/ruby/gems/2.3.0/gems/chef-12.16.42/lib/chef/provider/apt_repository.rb:59:in `block in <class:AptRepository>'
      
      apt_update("beats") do
        action [:nothing]
        ignore_failure true
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        declared_type :apt_update
        cookbook_name "filebeat"
      end
      
      Platform:
      ---------
      x86_64-linux
      
  
  * apt_preference[filebeat] action add[2017-02-01T16:28:01+00:00] WARN: Replacing filebeat.pref with filebeat.pref in /etc/apt/preferences.d/

    * directory[/etc/apt/preferences.d] action create (up to date)
    * file[cleanup_filebeat.pref] action delete (skipped due to only_if)
    * file[cleanup_filebeat] action delete (up to date)
    * file[/etc/apt/preferences.d/filebeat.pref] action create (up to date)
     (up to date)
  * apt_package[filebeat] action install
    * No candidate version available for filebeat
    ================================================================================
    Error executing action `install` on resource 'apt_package[filebeat]'
    ================================================================================
    
    Chef::Exceptions::Package
    -------------------------
    No candidate version available for filebeat
    
    Resource Declaration:
    ---------------------
    # In /root/.chef/local-mode-cache/cache/cookbooks/filebeat/recipes/install_package.rb
    
     57: package 'filebeat' do # ~FC009
     58:   version version_string
     59:   options node['filebeat']['apt']['options'] if node['filebeat']['apt']['options'] && node['platform_family'] == 'debian'
     60:   notifies :restart, 'service[filebeat]' if node['filebeat']['notify_restart'] && !node['filebeat']['disable_service']
     61:   flush_cache(:before => true) if node['platform_family'] == 'rhel'
     62:   allow_downgrade true if node['platform_family'] == 'rhel'
     63: end
    
    Compiled Resource:
    ------------------
    # Declared in /root/.chef/local-mode-cache/cache/cookbooks/filebeat/recipes/install_package.rb:57:in `from_file'
    
    apt_package("filebeat") do
      package_name "filebeat"
      action [:install]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :package
      cookbook_name "filebeat"
      recipe_name "install_package"
      version "5.2.0"
      options "-o Dpkg::Options::='--force-confnew'"
    end
    
    Platform:
    ---------
    x86_64-linux
    

Running handlers:
[2017-02-01T16:28:03+00:00] ERROR: Running exception handlers
Running handlers complete
```

**Comment by kmcarpenter** [comment]
In particular, it looks like apt is unable to even download the list of packages from elastic.co

**Comment by kmcarpenter** [comment]
This seems like it's related to me missing the following package: apt-transport-https

**Comment by vkhatri** [comment]
Fixed in v0.4.7. Thank you for sharing the issue details.
